main.cpp

```diff
@@ -613,10 +613,34 @@ static int parse_confs(ConfFile &conf)
         {"RetryTimeout",    false,  ConfFile::parse_i,          OPTIONS_TABLE_STRUCT_FIELD(option_tcp, timeout)},
     };
 
-    ret = conf.extract_options("General", option_table, ARRAY_SIZE(option_table), &opt);
+    static struct options options = {
+            .endpoints = nullptr,
+            .conf_file_name = nullptr,
+            .conf_dir = nullptr,
+            .tcp_port = ULONG_MAX,
+            .report_msg_statistics = false,
+            .logs_dir = nullptr,
+            .debug_log_level = (int)Log::Level::INFO,
+            .mavlink_dialect = Auto
+    };
+    ret = conf.extract_options("General", option_table, ARRAY_SIZE(option_table), &options);
     if (ret < 0)
         return ret;
 
+    // Merge general options giving preference to those from command line,
+    // when different from default
+    if (opt.tcp_port == ULONG_MAX)
+        opt.tcp_port = options.tcp_port;
+    if (!opt.report_msg_statistics)
+        opt.report_msg_statistics = options.report_msg_statistics;
+    if (opt.logs_dir == nullptr && options.logs_dir != nullptr)
+        opt.logs_dir = strdup(options.logs_dir);
+    if (opt.debug_log_level != (int)Log::Level::INFO)
+        opt.debug_log_level = options.debug_log_level;
```

Comment:
ugh... I don't want to do this for each and every option. You need to defer the command line parsing (probably with the exception of -c) to be done after the configuration files so it overrides them.